### PR TITLE
Update and expand DAGDependency drawer

### LINKED ISSUE #9066: Better DAGDependency visualization
## Description

We need a better way to visualize `DAGDependency` objects, so that we can understand from the picture on which qubits each gate acts on.

As an example, consider the following:

```
from qiskit.circuit import QuantumCircuit
from qiskit.dagcircuit import DAGCircuit, DAGDependency
from qiskit.converters import circuit_to_dag, circuit_to_dagdependency
from qiskit.visualization import dag_drawer

qc = QuantumCircuit(4)
qc.h(0)
qc.cx(0, 1)
qc.cx(0, 2)
qc.cx(0, 3)
qc.x(3)
qc.h(3)
```

Drawing this `QuantumCircuit` with
```
qc.draw()
```
produces
image
from which it's very clear to understand on which qubits each gate acts on.

Converting `QuantumCircuit` to `DAGCircuit` and drawing it with
```
dag = circuit_to_dag(qc)
dag_drawer(dag)
```
produces
image
and again it's easy to figure out on which qubits each gate acts on.

However, converting `QuantumCircuit` to `DAGDependency` and drawing it with
```
dagdep = circuit_to_dagdependency(qc)
dag_drawer(dagdep)
```
produces
image
from which it's almost impossible to figure out which gate is which, or what set of qubits it acts on.

## Request

It would be nice if for each gate in the last picture we could explicitly see the qubits it acts on.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #9066

### Details and comments

This PR updates the `DAGDependency` section of the `qiskit/visualization/dag_drawer.py`.
- The `node.qargs` and `node.cargs` have been added to each node display to indicate which qubits and clbits the instruction acts on.
- If an instruction has a condition, it will be displayed
- The colors have been changed to more closely match those of the `DAGCircuit` drawings.

This circuit
```
bits = [Qubit(), Clbit()]
qr = QuantumRegister(4, 'qr')
cr = ClassicalRegister(4, 'cr')
qc = QuantumCircuit(qr, bits, cr)
qc.h(0)
qc.cx(0, 1)
qc.cx(0, 2)
qc.cx(0, 3)
qc.x(3).c_if(cr[1], 1)
qc.h(3)
qc.x(4)
qc.barrier(0, 1)
qc.measure(0, 0)
print(qc)
dag = circuit_to_dagdependency(qc)
dag.draw()
```
produces this circuit drawing
```
      ┌───┐                 ░       ┌─┐
qr_0: ┤ H ├──■────■────■────░───────┤M├
      └───┘┌─┴─┐  │    │    ░       └╥┘
qr_1: ─────┤ X ├──┼────┼────░────────╫─
           └───┘┌─┴─┐  │    ░        ║ 
qr_2: ──────────┤ X ├──┼─────────────╫─
                └───┘┌─┴─┐┌───┐┌───┐ ║ 
qr_3: ───────────────┤ X ├┤ X ├┤ H ├─╫─
      ┌───┐          └───┘└─╥─┘└───┘ ║ 
   4: ┤ X ├─────────────────╫────────╫─
      └───┘                 ║        ║ 
   0: ══════════════════════■════════╩═
                                       
cr: 4/═════════════════════════════════
                                       
```
and this graph drawing
image

### CODE REVIEW

File: test/python/visualization/references/dag_no_reg.png
Comment (jakelishman) [inline test/python/visualization/references/dag_no_reg.png:1]: This one isn't a `DAGDependency`, right?

**Review by jakelishman — COMMENTED** [review]
In general, this looks a completely clear win in usability to me.  I haven't used `DAGDependency` much though, so it'd be good to get a second opinion.  @alexanderivrii, does the new output format look good to you here?

Can we have a "feature" release note for this improvement to the drawers as well?

File: test/python/visualization/references/dag_no_reg.png
Comment (enavarro51) [reply test/python/visualization/references/dag_no_reg.png:1]: Right. I was going to use a 1% diff on the image tests, but this one was at 6%, so I thought I'd update the reference image. Turns out after doing this it was still at 6% and the DAGDependency image was at 8% on the CI run. So I don't know what the difference is between the fonts, image libraries, etc. between my local and CI. I have the recommended version of graphviz. So we could just leave the old one.

**Comment by enavarro51** [comment]
I'll add a reno.

File: test/python/visualization/references/dag_no_reg.png
Comment (jakelishman) [reply test/python/visualization/references/dag_no_reg.png:1]: I mean, the new one looks more like it's intended and it passes CI, so I'm fine leaving it.

**Comment by enavarro51** [comment]
Reno added in https://github.com/Qiskit/qiskit/pull/11103/commits/53381b3c42e1755a3c1d5a5ff48f01a97c5e1a0a.

**Comment by alexanderivrii** [comment]
Thanks @enavarro51, I really like the new output. The code changes look good as well.

**Review by alexanderivrii — APPROVED** [review]

**Pull request merged**